assets/src/components/ladder.tsx

```diff
@@ -39,38 +46,70 @@ const Ladder = ({
   ladderDirection,
   selectedVehicleId,
 }: Props) => {
-  const dispatch = useContext(DispatchContext)
-  const timepointSpacingY =
-    (HEIGHT - MARGIN_TOP_BOTTOM * 2) / (timepoints.length - 1)
-
   const orderedTimepoints: Timepoint[] =
     // Use slice to make a copy of the array before destructively reversing
     ladderDirection === LadderDirection.OneToZero
       ? timepoints.slice().reverse()
       : timepoints
 
+  const timepointSpacingY: TimepointSpacingYFunc = (height: number): number =>
+    (height - MARGIN_TOP_BOTTOM * 2) / (timepoints.length - 1)
   const timepointStatusY = timepointStatusYFromTimepoints(
     timepoints,
     timepointSpacingY
   )
+
   const { ladderVehicles, widthOfLanes } = ladderVehiclesFromVehicles(
     vehicles,
     ladderDirection,
     timepointStatusY
   )
 
   const width = 120 + 2 * widthOfLanes
-  // (0, 0) is in the center of the first timepoint
-  const viewBox = [-width / 2, -MARGIN_TOP_BOTTOM, width, HEIGHT].join(" ")
 
   return (
-    <svg className="m-ladder" height={HEIGHT} width={width} viewBox={viewBox}>
+    <div className="m-ladder">
+      <LadderSvg
+        timepoints={orderedTimepoints}
+        ladderVehicles={ladderVehicles}
+        selectedVehicleId={selectedVehicleId}
+        width={width}
+        timepointSpacingY={timepointSpacingY}
+        timepointStatusY={timepointStatusY}
+      />
+    </div>
+  )
+}
+
+const LadderSvg = ({
```

Comment:
If all of this stays in one component, I think a few things would be a little simpler. E.g. `timepointSpacingY` could be a variable calculated after `height = useState()` instead of being a function with its own `TimepointSpacingYFunc`.

What's the benefit of splitting the `LadderSvg` stuff into a separate component, and is it worth the overhead + extra complexity?

Reply:
Yep, this is better. That was fallout from one of the several approaches I tried.